pkg/actuators/machine/actuator.go

```diff
@@ -0,0 +1,221 @@
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog"
+
+	osclientset "github.com/openshift/client-go/config/clientset/versioned"
+	ovirtconfigv1 "github.com/openshift/cluster-api-provider-ovirt/pkg/apis/ovirtprovider/v1beta1"
+	ovirtC "github.com/openshift/cluster-api-provider-ovirt/pkg/clients/ovirt"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	TimeoutInstanceCreate       = 5 * time.Minute
+	RetryIntervalInstanceStatus = 10 * time.Second
+)
+
+// ActuatorParams holds parameter information for Actuator
+type ActuatorParams struct {
+	Namespace      string
+	Client         client.Client
+	KubeClient     *kubernetes.Clientset
+	Scheme         *runtime.Scheme
+	MachinesClient v1beta1.MachineV1beta1Interface
+	EventRecorder  record.EventRecorder
+}
+
+type OvirtActuator struct {
+	params          ActuatorParams
+	scheme          *runtime.Scheme
+	client          client.Client
+	KubeClient      *kubernetes.Clientset
+	machinesClient  v1beta1.MachineV1beta1Interface
+	EventRecorder   record.EventRecorder
+	ovirtConnection *ovirtsdk.Connection
+	OSClient        osclientset.Interface
+}
+
+func NewActuator(params ActuatorParams) (*OvirtActuator, error) {
+	config := ctrl.GetConfigOrDie()
+	osClient := osclientset.NewForConfigOrDie(rest.AddUserAgent(config, "cluster-api-provider-ovirt"))
+
+	return &OvirtActuator{
+		params:          params,
+		client:          params.Client,
+		machinesClient:  params.MachinesClient,
+		scheme:          params.Scheme,
+		KubeClient:      params.KubeClient,
+		EventRecorder:   params.EventRecorder,
+		ovirtConnection: nil,
+		OSClient:        osClient,
+	}, nil
+}
+
+// Create creates a machine(oVirt VM) and is invoked by the machine controller.
+func (actuator *OvirtActuator) Create(ctx context.Context, machine *machinev1.Machine) error {
+	providerSpec, err := ovirtconfigv1.ProviderSpecFromRawExtension(machine.Spec.ProviderSpec.Value)
+	if err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.InvalidMachineConfiguration(
+			"Cannot unmarshal machineProviderSpec field: %v", err))
+	}
+
+	connection, err := actuator.getConnection()
+	if err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.InvalidMachineConfiguration(
+			"failed to create connection to oVirt API: %v", err))
+	}
+
+	ovirtClient := ovirtC.NewOvirtClient(connection)
+	mScope := newMachineScope(ctx, ovirtClient, actuator.client, actuator.machinesClient, machine, providerSpec)
+
+	if err := mScope.create(); err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.CreateMachine(
+			"Error creating Machine %v", err))
+	}
+
+	if err := mScope.patchMachine(ctx, conditionSuccess()); err != nil {
+		return actuator.handleMachineError(machine, "Create", apierrors.CreateMachine(
+			"Error patching Machine %v", err))
+		return err
```

Comment:
Do not return errors without context, wrap errors instead.

Reply:
removed that part, 2 returns once after the other just a silly bug